test/jdk/java/lang/String/RegionMatches.java

```diff
@@ -23,18 +23,36 @@
 
 /**
  * @test
- * @bug 4016509
- * @summary test regionMatches corner case
+ * @bug 4016509 8316879
+ * @summary test regionMatches corner cases
+ * @run junit RegionMatches
  */
 
+import java.io.UnsupportedEncodingException;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class RegionMatches {
 
-  public static void main (String args[]) throws Exception {
-      String s1="abc";
-      String s2="def";
+  private final String s1_LATIN1 = "abc";
+  private final String s2_LATIN1 = "def";
 
-      if (!s1.regionMatches(0,s2,0,Integer.MIN_VALUE))
-          throw new RuntimeException("Integer overflow in RegionMatches");
+  private final byte[] b1_UTF16 = new byte[]{0x04, 0x3d, 0x04, 0x30, 0x04, 0x36, 0x04, 0x34};
+  private final byte[] b2_UTF16 = new byte[]{0x04, 0x32, 0x00, 0x20, 0x04, 0x41, 0x04, 0x42};
```

Comment:
For strings, the \uxxxx version would be preferred; it is clearer that what the character is and there is less of a chance that the UTF encoding has a mistake.

Reply:
Done!